File: MythonJava/src/sample/types/List.java
package sample.types;

import java.util.ArrayList;

public class List extends Sequence {

    private ArrayList<Type> types;

    public List(ArrayList<Type> types) {
        this.types = types;
    }

    public List(boolean isTemp, ArrayList<Type> types) {
        super(isTemp);
        this.types = types;
    }

    public void append(Type type) {
        this.types.add(type);
    }

    @Override
    public boolean isPrintable() {
        return true;
    }

    @Override
    public String toString() {
        if (this.types.size() == 0)
            return "[]";

        String s = "[";

        for (int i = 0; i < this.types.size() - 1; i++)
            s += this.types.get(i).toString() + ", ";
        s += this.types.get(this.types.size() - 1) + "]";
        return s;
    }

    @Override
    public String getType() {
        return "<type 'list'>";
    }

    public int len() {
        return this.types.size();
    }
}

File: MythonC++/Mython/List.cpp
#include "List.h"

List::List(vector<Type*> values, string originalList) : Sequence(false)
{
	this->values = values;
}

List::List(bool isTemp, vector<Type*> values, string originalList) : Sequence(isTemp)
{
	this->values = values;
}

const bool List::IsPrintable()
{
	return true;
}

/// <summary>
/// This function appends a value to the list.
/// </summary>
/// <param name="value"></param>
void List::Append(Type * t)
{
	this->values.push_back(t);
}

List* List::Clone()
{
	return new List(this->_isTemp, this->values, this->originalList);
}

string List::GetOriginalList()
{
	return this->originalList;
}

string List::GetType()
{
	return TYPE_LIST;
}

int List::Len()
{
	return this->values.size();
}

string List::ToString()
{
	if (this->values.empty())
		return "[]";

	string s = "[";

	for (int i = 0; i < this->values.size() - 1; i++)
	{
		s += this->values[i]->ToString();
		s += ", ";
	}
	s += this->values[this->values.size() - 1]->ToString();
	s += "]";
	return s;
}
